tracing-journald/src/lib.rs

```diff
@@ -174,12 +179,39 @@ where
             self.field_prefix.as_ref().map(|x| &x[..]),
         ));
 
-        // What could we possibly do on error?
-        #[cfg(unix)]
-        let _ = self.socket.send(&buf);
+        // At this point we can't handle the error anymore so just ignore it.
+        let _ = send_payload(&self.socket, &buf);
     }
 }
 
+#[cfg(not(target_os = "linux"))]
+fn send_payload(socket: &UnixDatagram, payload: &[u8]) -> std::io::Result<()> {
+    Err(std::io::Error::new(
+        std::io::ErrorKind::Unsupported,
+        "journald not supported on non-Linux OS",
+    ))
+}
+
+#[cfg(target_os = "linux")]
+fn send_payload(socket: &UnixDatagram, payload: &[u8]) -> std::io::Result<usize> {
```

Comment:
nit: `std::io` is imported in this module, so these could just be:
```suggestion
    Err(io::Error::new(
        io::ErrorKind::Unsupported,
        "journald not supported on non-Linux OS",
    ))
}

#[cfg(target_os = "linux")]
fn send_payload(socket: &UnixDatagram, payload: &[u8]) -> io::Result<usize> {
```

Reply:
Thanks.  I had to refactor this anyway, because I missed that `UnixDatagram` doesn't exist on non-Unix systems.